ansible/roles/bootstrap/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+- name: Ensures bootkube dir exists
+  file: path=/tmp/bootkube state=directory
+
+- name: Extract bootkube binaries
+  unarchive:
+    src: https://github.com/kubernetes-incubator/bootkube/releases/download/v0.3.9/bootkube.tar.gz
+    dest: /tmp/bootkube
+    remote_src: True
+
+- name: Render bootkube manifests
+  command: /tmp/bootkube/bin/linux/bootkube render --asset-dir=/tmp/bootkube/assets --experimental-self-hosted-etcd --etcd-servers=http://10.3.0.15:2379 --api-servers=https://{{ inventory_hostname }}:443
```

Comment:
For now the self-hosted implementation in kubeadm is only a first shot and limited. It definitely makes sense to move to kubeadm at a later point, because this is where development is happening. When it's broken down in different components (probably with 1.7).

bootkube itself will vanish in the future: 

> "We hope that either bootkube is superseded by the introduction of a mechanism to push manifests into a kubelet, which rectify with the API later, and are checkpointed by the kubelet. Or, merge the project with hyperkube if we decide that the kubelet approach isn't reasonable. While this plays out we think the project is useful for keeping the discussion around self-hosted moving forward. Further discussion can be found on this issue: https://github.com/coreos/bootkube/issues/97"